common/src/components/accordion/Section.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from "react";
+import { ChevronDownIcon, ChevronRightIcon } from "@heroicons/react/solid";
+
+export interface ExternalSectionProps {
+  title: string;
+  subtitle?: string;
+  Icon?: React.ComponentType<React.SVGProps<SVGSVGElement>>;
+}
+
+interface InternalSectionProps extends ExternalSectionProps {
+  open?: boolean;
+}
+
+export const Section: React.FC<InternalSectionProps> = ({
+  open,
+  title,
+  subtitle,
+  Icon,
+  children,
+}) => {
+  const contentRef = React.useRef(null);
+  const [isOpen, setIsOpen] = useState(open);
+
+  const handleOpen = () => {
+    setIsOpen((prev: boolean | undefined) => !prev);
+  };
+
+  return (
+    <div
+      data-h2-margin="b(top, xxs)"
+      data-h2-shadow="b(m)"
+      data-h2-radius="b(none, s, s, none)"
+      data-h2-overflow="b(all, hidden)"
+      data-h2-border="b([light]primary, left, solid, m)"
+    >
+      <button
+        type="button"
+        data-h2-text-align="b(left)"
+        data-h2-bg-color="b(white)"
+        data-h2-padding="b(top-bottom, s) b(right, m) b(left, s)"
+        data-h2-width="b(100)"
+        onClick={() => handleOpen()}
+        {...(isOpen && children
+          ? { "data-h2-border": "b(lightpurple, left, solid, m)" }
+          : { "data-h2-border": "b(darkpurple, left, solid, m)" })}
+        style={{ borderTop: "none", borderRight: "none", borderBottom: "none" }}
+      >
+        <div data-h2-flex-grid="b(middle, expanded, flush, s)">
+          <span>
+            {isOpen ? (
+              <ChevronDownIcon height="20" />
+            ) : (
+              <ChevronRightIcon height="20" />
+            )}
+          </span>
+          <div data-h2-flex-item="b(auto)" data-h2-text-align="b(left)">
+            <p data-h2-margin="b(all, none)" data-h2-font-size="b(h5)">
+              {title}
+            </p>
+            <p data-h2-margin="b(top, xxs) b(bottom, none)">{subtitle}</p>
+          </div>
+          <div data-h2-flex-item="b(content)">
+            <span className="icon" data-h2-text-align="b(left)">
+              {Icon && <Icon height="20px" width="20px" />}
+            </span>
+          </div>
+        </div>
+      </button>
+      <div
+        ref={contentRef}
+        data-h2-border="b(lightpurple, left, solid, m)"
+        id="content"
+      >
+        {isOpen && (
+          <div data-h2-padding="b(top, none) b(right, l) b(bottom, m) b(left, xl)">
+            <hr data-h2-margin="b(top, none) b(bottom, s)" />
+            <p data-h2-margin="b(all, none)">{isOpen && children} </p>{" "}
```

Comment:
Since the section is already conditional on isOpen, we can just have `{children}` on this line.

Also, there's no guarantee that children is a text block. It could be another complex component, even a form. Maybe try to check if children is a string. If it is a string, wrap it in a <p> element. If its not (probably another component) then wrap it in a <div> (if we want to guarantee the margin) or leave it bare instead.

Reply:
removed the check but it just works fine with plain string or component.